crates/services/p2p/src/p2p_service.rs

```diff
@@ -609,11 +609,22 @@ impl<Codec: NetworkCodec> FuelP2PService<Codec> {
                                     );
                                 }
                             }
+                            (
+                                Some(ResponseChannelItem::Transactions2(channel)),
+                                Ok(ResponseMessage::Transactions2(transactions)),
+                            ) => {
+                                if channel.send(transactions).is_err() {
+                                    debug!(
+                                        "Failed to send through the channel for {:?}",
+                                        request_id
+                                    );
+                                }
+                            }
                             (
                                 Some(ResponseChannelItem::SealedHeaders(channel)),
                                 Ok(ResponseMessage::SealedHeaders(headers)),
                             ) => {
-                                if channel.send((peer, headers)).is_err() {
+                                if channel.send(headers).is_err() {
```

Comment:
In this PR, we don't send the peer in the response because the peer is now sent in the request. But if I can revert this back to random peer selection, this may change again.

Reply:
Updated